exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -252,6 +252,21 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 		return err
 	}
 
+	// Ensure that the kubernetes service of remote cluster has been created before apply ClusterResourceSet.
+	err = remoteClient.Get(ctx, types.NamespacedName{
+		Namespace: metav1.NamespaceDefault,
+		Name:      "kubernetes",
+	}, &corev1.Service{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		log.Error(err, "failed to get kubernetes service", "Cluster", util.ObjectKey(cluster).String())
+		return err
+	}
+
+	if apierrors.IsNotFound(err) {
+		log.Info("Wait for the Service kubernetes to be created")
```

Comment:
As above - let's wrap these errors to add context, but there's no need for a separate log call.